docs, kpr, maglev: Move Maglev out of beta

borkmann opened the pull request:
see commit msg.

aditighag reviewed (commented):
I don't have much experience working with Maglev, but I see these issues currently open that stand out - 
https://github.com/cilium/cilium/issues/19260
https://github.com/cilium/cilium/issues/18744 (needs more triaging/information)
https://github.com/cilium/cilium/issues/19453 (possibly kernel regression?)

@borkmann Do you mind taking a look at these issues again against the backdrop of promoting Maglev to stable?

brb reviewed (approved):
:+1: Judging from user reports the feature has seen a fair amount of usage.

borkmann committed:
cilium, docs: Move Maglev out of beta state

It has seen enough production exposure over the years and there are no
known issues atm.

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

xmulligan commented on Documentation/community/roadmap.rst:
Should we also add this to the table above under "eBPF networking"?

borkmann replied:
Given this is a subfeature of `Kube-proxy Replacement` and `Load Balancing`, I don't think it would be the right place to list there.
